src/WinRT.Runtime/TypeNameSupport.cs

```diff
@@ -425,22 +419,41 @@ private static bool TryAppendTypeName(Type type, StringBuilder builder, TypeName
 #else
             if (type.IsSZArray)
 #endif
-            {
-                builder.Append("Windows.Foundation.IReferenceArray`1<");
-                if (TryAppendTypeName(type.GetElementType(), builder, flags & ~TypeNameGenerationFlags.GenerateBoxedName))
-                {
-                    builder.Append('>');
-                    return true;
+            {
+                var elementType = type.GetElementType();
+                if (elementType.ShouldProvideIReference())
+                {
+                    builder.Append("Windows.Foundation.IReferenceArray`1<");
+                    if (TryAppendTypeName(elementType, builder, flags & ~TypeNameGenerationFlags.GenerateBoxedName))
+                    {
+                        builder.Append('>');
+                        return true;
+                    }
+                    return false;
                 }
-                return true;
+                else
+                {
+                    return false;
+                }
+            }
+
+            if ((flags & TypeNameGenerationFlags.GenerateBoxedName) != 0 && type.ShouldProvideIReference())

```

Comment:
For a non-projected enum, if type.ShouldProvideIReference() returns false, then does it still go to TryAppendSimpleTypeName below and have its type name appended?

Reply:
Yes in the next if statement.